src/main/java/org/apache/datasketches/memory/Memory.java

```diff
@@ -17,36 +17,23 @@
  * under the License.
  */
 
-package org.apache.datasketches.memory;
 
-import static org.apache.datasketches.memory.Util.negativeCheck;
-import static org.apache.datasketches.memory.Util.nullCheck;
-import static org.apache.datasketches.memory.Util.zeroCheck;
+package org.apache.datasketches.memory;
 
 import java.io.File;
 import java.io.IOException;
 import java.nio.ByteBuffer;
 import java.nio.ByteOrder;
 import java.nio.channels.WritableByteChannel;
 
-/**
- * Provides read-only primitive and primitive array methods to any of the four resources
- * mentioned in the package level documentation.
- *
- * @author Roman Leventov
- * @author Lee Rhodes
- *
- * @see org.apache.datasketches.memory
- */
-public abstract class Memory extends BaseState {
+import org.apache.datasketches.memory.internal.MemoryImpl;
+import org.apache.datasketches.memory.internal.Utf8CodingException;
```

Comment:
Since this exception may be thrown whilst using a public method, it might be necessary to move this into the root package - otherwise a developer may not be able to explicitly catch this exception directly.

Reply:
Good catch! Thanks!